### DISCUSSION TITLE
How to force a line break in ui.table

### DISCUSSION THREAD

**Discussion by steeeef99** [zauberzeug/nicegui#2411] [Q&A] [answered]
### Question

Hi!
Is it possible to force a line break in ui.table?

I have a table and in some cells there is a text, that contains \n. I would like to have a line break in the table-cell, whenever there is a \n. The separator don't have to be a \n it could also be a comma or a blank.

Does anyone have an idea how to do that?

Here is what I tried:
- add a class to ui.table: word-break
- add a slot to my table:
```py
table.add_slot('body-cell', r'''
                    <q-td :props="props">
                    {{ props.row.name.replace(/\n/g, '<br>') }}
                    </q-td>''')
```

Thanks in advance, stefan

**Comment by falkoschindler** [comment, marked as answer]
Hi @steeeef99,

To define the _HTML_ of the table cell (instead of the inner text), you can use Vue's `v-html` attribute like this:
```py
table.add_slot('body-cell-name', r'''<q-td :props="props" v-html="props.value.replace(/\n/g, '<br>')" />''')
```

**Reply by steeeef99** [reply]
Hi @falkoschindler,
thanks for the quick reply! It worked :)

**Comment by steeeef99** [comment]
Hi @falkoschindler,
I found another way, that may be interesting:

```
table.add_slot('body-cell', r'''
            <td :props="props" :style="{'white-space':'pre-line'}">{{ props.value }}</td>
        ''')
```

Every '\n' in the text will be interpreted as a line break in the table afterwards.

best regards Stefan

**Reply by erw-emd-dk** [reply]
Thanks, this helped me, but I used `'white-space':'pre'` instead of `'white-space':'pre-line'` to preserve linebreaks but avoid wrapping.
